### PR TITLE
feat(sensenova_u1): add prefix-identity hash production and its suites

### PR DESCRIPTION
**Pull request by 0z5a**
## What this adds

`DiffusionKVRequest.block_hashes` documents a canonical block hash, but nothing
populated it. This adds the production module that computes it, plus the CPU
suites that pin its contract.

`vllm_omni/diffusion/models/sensenova_u1/prefix_identity.py` derives a block hash
that separates every layer of the identity:

- branch
- weight epoch
- TP rank topology
- ancestor hash
- the block's own tokens

A layer that silently fails to separate two computations lets one request reuse
another's KV, so each layer is tested on its own.

| file | kind |
| --- | --- |
| `vllm_omni/diffusion/models/sensenova_u1/prefix_identity.py` | new module (hash production) |
| `tests/.../test_prefix_identity.py` | module tests, incl. driving the real `KVCacheManager` |
| `tests/.../test_sensenova_u1_prefix_contract.py` | the conservative reuse cut |
| `tests/.../test_sensenova_u1_prefix_matrix.py` | boundary / partial-hit / branch matrix |
| `tests/.../test_sensenova_u1_prefix_parity.py` | identity and parity |
| `tests/.../test_sensenova_u1_loader_routing.py` | routing over real checkpoint names |
| `tests/.../test_sensenova_u1_tp_structural.py` | TP sharding and degree legality |

## Test result

CPU only, no GPU. The suite is gated on the real checkpoint so the loader audit
runs over real tensor names rather than a fixture.

| suite | tests | result |
| --- | --- | --- |
| `test_prefix_identity.py` | 27 | pass |
| `test_sensenova_u1_prefix_contract.py` | 16 | pass |
| `test_sensenova_u1_prefix_matrix.py` | 26 | pass |
| `test_sensenova_u1_prefix_parity.py` | 10 | pass |
| `test_sensenova_u1_tp_structural.py` | 16 | pass |
| `test_sensenova_u1_loader_routing.py` | 12 | pass |
| **total** | **107** | **107 passed, 0 failed, 0 skipped** |

```
$ SENSENOVA_MODEL_DIR=<SenseNova-U1.5-8B-MoT> python -m pytest \
    tests/diffusion/models/sensenova_u1/test_prefix_identity.py \
    tests/diffusion/models/sensenova_u1/test_sensenova_u1_{loader_routing,prefix_contract,prefix_matrix,prefix_parity,tp_structural}.py
107 passed, 16 warnings in 0.20s
```

Without `SENSENOVA_MODEL_DIR` the loader audit self-skips with an explicit reason
(`set SENSENOVA_MODEL_DIR to audit every real checkpoint name`), so the other 106
still run in a bare checkout.

Evidence beyond the raw pass count, from the accompanying analysis:

| check | cases | mismatches |
| --- | --- | --- |
| mask semantics vs an independent dense reference | 1,092 vectors | 0 |
| plan oracle vs mask-derived safe cut | 12,116,880 | 0 |
| cut oracle closure and safety | 123,015 | 0 |
| identity vs the real vLLM `KVCacheManager` | 14 | 0 |
| loader routing over all real tensor names | 1,116 names | 0 mis-routes |

The suites are load-bearing rather than vacuous: **8/8 semantic mutants** are
caught by `mutation_check.py` (each mutant disables one identity layer and the
corresponding assertion fails).

## Scope

Additive. No tracked file is modified, and the module imports no `vllm`, so the
suites run in a CPU-only environment. The one failure in the wider
`sensenova_u1` directory
(`test_sensenova_u1_logits_processor.py::test_logits_processor_head_dtype_under_diffusion_shim`)
is pre-existing and in a file this change does not touch.

## Provenance

Every blob in this commit is byte-identical to the tree the results above were
produced from, verified by comparing `git hash-object` on the tested checkout
with the committed blob shas: `912303b8`, `9ebc1ae6`, `50b7eca9`, `e41be073`,
`075c1ebd`, `36eff0c7`, `3c2b403e`.

Refs #7716 (B3), #7715 (B5), #7598.

### CODE REVIEW

**Comment by 0z5a** [comment]
Superseded: splitting this into two focused PRs so B3 and B5 can be reviewed and land independently.

- B3 (AR prefix-reuse contract / identity) — refs #7716
- B5 (TP>1 structural evidence) — refs #7715

The two halves are independent: nothing under `tests/.../sensenova_u1/` added by B5 imports `prefix_identity`, so B5 does not need B3 to land first. The branch `feat/sensenova-u1-prefix-identity` is left in place for reference.
